### PR TITLE
Geowave 647 9.1

### PR DESCRIPTION
**Pull request by rwgdrummer**

### CODE REVIEW

File: analytics/api/src/main/java/mil/nga/giat/geowave/analytic/partitioner/OrthodromicDistancePartitioner.java [hunk 1]
```diff
@@ -70,7 +70,7 @@
 	private String crsName;
 	private transient CoordinateReferenceSystem crs = null;
 	private transient GeometryCalculations calculator;
-	private DimensionExtractor<T> dimensionExtractor;
+	protected DimensionExtractor<T> dimensionExtractor;
```
Comment (rwgdrummer) [inline analytics/api/src/main/java/mil/nga/giat/geowave/analytic/partitioner/OrthodromicDistancePartitioner.java]: Will make sense in the next commit.  It slipped in this one.  I just noticed it.  Recommend letting it go through.

File: extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/GeotoolsFeatureDataAdapter.java [hunk 2]
```diff
@@ -14,4 +14,6 @@
 	public SimpleFeatureType getType();
 
 	public TimeDescriptors getTimeDescriptors();
+
+	public boolean hasTemporalConstraints();
```
Comment (rwgdrummer) [inline extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/GeotoolsFeatureDataAdapter.java]: Rermoves  a 'subclass'   reference else where.

File: extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/KryoFeatureDataAdapter.java [hunk 3]
```diff
@@ -1,280 +0,0 @@
-package mil.nga.giat.geowave.adapter.vector;
```
Comment (rwgdrummer) [inline extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/KryoFeatureDataAdapter.java]: Removed...not used.

File: extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/WholeFeatureDataAdapter.java [hunk 4]
```diff
@@ -97,4 +259,114 @@ public AdapterPersistenceEncoding encode(
...
+	@Override
+	protected byte[] defaultTypeDataToBinary() {
+		final String encodedType = DataUtilities.encodeType(featureType);
+		final String typeName = featureType.getTypeName();
+		final byte[] typeNameBytes = StringUtils.stringToBinary(typeName);
+		final byte[] encodedTypeBytes = StringUtils.stringToBinary(encodedType);
+
+		byte[] attrBytes = new byte[0];
+
+		final SimpleFeatureUserDataConfigurationSet userDataConfiguration = new SimpleFeatureUserDataConfigurationSet();
```
Comment (rwgdrummer) [inline extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/WholeFeatureDataAdapter.java]: Visibility per cell not used.

File: extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/index/ChooseBestMatchIndexQueryStrategy.java [hunk 5]
```diff
@@ -68,6 +69,10 @@ public boolean hasNext() {
 						nextIdx = (PrimaryIndex) nextChoosenIdx;
 						if (nextIdx.getIndexStrategy().getOrderedDimensionDefinitions().length == 0) continue;
 						final List<MultiDimensionalNumericData> constraints = query.getIndexConstraints(nextIdx.getIndexStrategy());
+
```
Comment (rwgdrummer) [inline extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/index/ChooseBestMatchIndexQueryStrategy.java]: Choose best match only uses stats.
Some indices do not have stats (liek NullIndex)

File: extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/plugin/GeoWaveGTDataStore.java [hunk 6]
```diff
@@ -143,14 +143,6 @@ public DataStatisticsStore getDataStatisticsStore() {
 		return dataStatisticsStore;
 	}
 
-	protected List<PrimaryIndex> getWriteIndices(
```
Comment (rwgdrummer) [inline extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/plugin/GeoWaveGTDataStore.java]: Not needed...just a silly wrapper around getPrefrredIndices

**Comment by rfecher** [comment]
so this is just a cherry-pick?  it seems ok, a lot of changes that I don't necessarily need to review again assuming it is just a cherry-pick?

**Comment by rwgdrummer** [comment]
Correct just cherry picking.  The change is that I removed  per attribute visibility from the whole feature data adapter.  It was not not used.  Also, I collapsed the Kryo adapter into the whole feature adapter.  The kryo class was not serving any purpose.

**Pull request merged**